src/transformers/models/jina_bert/modular_jina_bert.py

```diff
@@ -0,0 +1,1169 @@
+                if self.config.add_cross_attention:
+                    all_cross_attentions = all_cross_attentions + (layer_outputs[2],)
+
+        if output_hidden_states:
+            all_hidden_states = all_hidden_states + (hidden_states,)
+
+        if not return_dict:
+            return tuple(
+                v
+                for v in [
+                    hidden_states,
+                    next_decoder_cache,
+                    all_hidden_states,
+                    all_self_attentions,
+                    all_cross_attentions,
+                ]
+                if v is not None
+            )
+        return BaseModelOutputWithPastAndCrossAttentions(
+            last_hidden_state=hidden_states,
+            past_key_values=next_decoder_cache,
+            hidden_states=all_hidden_states,
+            attentions=all_self_attentions,
+            cross_attentions=all_cross_attentions,
+        )
+
+
+# class JinaBertLMPredictionHead(BertLMPredictionHead):
+#     def _tie_weights(self):
+#         raise AttributeError("Not needed for JinaBert")
+
+# class JinaBertOnlyMLMHead(BertOnlyMLMHead):
+#     def __init__(self, config):
+#         super().__init__()
+#         self.predictions = JinaBertLMPredictionHead(config)
+
+# class JinaBertPreTrainingHeads(BertPreTrainingHeads):
+#     def __init__(self, config):
+#         super().__init__()
+#         self.predictions = JinaBertLMPredictionHead(config)
+
+
+class JinaBertPreTrainedModel(BertPreTrainedModel):
+    config_class = JinaBertConfig
+    _supports_sdpa = None
+    _no_split_modules = ["JinaBertLayer"]
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, JinaBertEncoder):
+            module.gradient_checkpointing = value
+
+
+class JinaBertModel(BertModel):
+    """
+
+    The model can behave as an encoder (with only self-attention) as well as a decoder, in which case a layer of
+    cross-attention is added between the self-attention layers, following the architecture described in [Attention is
+    all you need](https://arxiv.org/abs/1706.03762) by Ashish Vaswani, Noam Shazeer, Niki Parmar, Jakob Uszkoreit,
+    Llion Jones, Aidan N. Gomez, Lukasz Kaiser and Illia Polosukhin.
+
+    To behave as an decoder the model needs to be initialized with the `is_decoder` argument of the configuration set
+    to `True`. To be used in a Seq2Seq model, the model needs to initialized with both `is_decoder` argument and
+    `add_cross_attention` set to `True`; an `encoder_hidden_states` is then expected as an input to the forward pass.
+    """
+
+    def __init__(self, config: JinaBertConfig, add_pooling_layer=True):
+        super().__init__(config)
+        self.config = config
+
+        self.emb_pooler = config.emb_pooler
+        self._name_or_path = config._name_or_path
+        if self.emb_pooler:
+            self.tokenizer = AutoTokenizer.from_pretrained(config._name_or_path)
```

Comment:
The [original implementation](https://huggingface.co/jinaai/jina-bert-v2-qk-post-norm/blob/main/modeling_bert.py#L1124) had a conditional import here.
That didn't work for me, so I moved the import to the [top](https://github.com/huggingface/transformers/pull/35320/files#diff-8fe98a0547962090c974964db874a463d2541284c1865dc2661ad9f5a9fb007cR28).